### PR TITLE
ci: Use corepack to control package manager versions

### PR DESCRIPTION
**Pull request by tylerbutler**
## Description

Updates the CI pipeline and readme instructions to use [corepack](https://nodejs.org/docs/latest-v16.x/api/corepack.html) to install and use the correct version of pnpm. The hash of the package manager is included to improve the security of the CI pipeline. If the hashes don't match when node downloads the package manager then the CI pipeline will fail at the "install and configure pnpm" step.

Example failing CI run:

https://dev.azure.com/fluidframework/public/_build/results?buildId=146662&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=996a989a-edd9-5084-d9de-d81350de1bfc

To determine the sha512 hash, I used Linux and ran `npm view pnpm@7.30.5 dist.tarball` to get the URL of the package tarball. Then I did the following:

```shell
wget https://registry.npmjs.org/pnpm/-/pnpm-7.30.5.tgz
sha512sum pnpm-7.30.5.tgz
```

### CODE REVIEW

**Comment by CraigMacomber** [comment]
> I used SHA1 since that's what `npm view pnpm@7.30.5 dist.shasum` provides.

SHA1 has been broken and shouldn't be used anymore. npm view reports:
```
> npm view pnpm@7.30.5

pnpm@7.30.5 | MIT | deps: none | versions: 948
...

dist
.tarball: https://registry.npmjs.org/pnpm/-/pnpm-7.30.5.tgz
.shasum: ba4f8057ab7b24db2ebe553808b95adb55b5330f
.integrity: sha512-kXiH7+iGhDcm3UVhjb4pzbRYljoT2KVR8WFL37b+c15F9CuaLau0RTozrXx/9snf1JEmGICjRnMM2XArmM01sg==
.unpackedSize: 13.0 MB

...
```

Can you use the sha512 instead?

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -26,8 +26,8 @@
 
 	// Invoke 'nvm' to install our preferred version of node, per the '.nvmrc' file
 	// located at the root of the ${workspaceFolder}.
-	// Also install pnpm as it is now required to install all packages
-	"postCreateCommand": ". /usr/local/share/nvm/nvm.sh; nvm install; npm i -g pnpm@7",
+	// Also run corepack enable to allow node to download and use the right version of pnpm.
```
Comment (CraigMacomber) [inline .devcontainer/devcontainer.json:29]: I don't see this running `corepack enable`. Does something else run that or is it missing?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -26,8 +26,8 @@
...
+	"postCreateCommand": ". /usr/local/share/nvm/nvm.sh; nvm install",
```
Comment (tylerbutler) [inline .devcontainer/devcontainer.json:30]:
```suggestion
	"postCreateCommand": ". /usr/local/share/nvm/nvm.sh; nvm install; corepack enable",
```

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (tylerbutler) [reply .devcontainer/devcontainer.json:29]: mistake; fixed.

File: tools/test-tools/package.json [hunk 3]
```diff
@@ -40,5 +40,6 @@
 		"prettier": "~2.6.2",
 		"rimraf": "^2.6.2",
 		"typescript": "~4.5.5"
-	}
+	},
+	"packageManager": "pnpm@7.30.5+sha1.ba4f8057ab7b24db2ebe553808b95adb55b5330f"
```
Comment (CraigMacomber) [inline tools/test-tools/package.json:44]: Does this work? I couldn't find any examples using sha512 with packageManager, but maybe this format works?

```suggestion
	"packageManager": "pnpm@7.30.5+sha512-kXiH7+iGhDcm3UVhjb4pzbRYljoT2KVR8WFL37b+c15F9CuaLau0RTozrXx/9snf1JEmGICjRnMM2XArmM01sg=="
```

Comment (tylerbutler) [reply tools/test-tools/package.json]: That form doesn't work unfortunately. I downloaded the tarball manually and used `sha512sum` which produced `917887efe886843726dd45618dbe29cdb458963a13d8a551f1614bdfb6fe735e45f42b9a2dabb4453a33ad7c7ff6c9dfd491261880a346730cd9702b98cd35b2`.

Trying that now.

**Review by CraigMacomber — APPROVED** [review]

File: tools/test-tools/package.json (hunk 3 above)
Comment (tylerbutler) [reply tools/test-tools/package.json]: Worked!

**Comment by alexvy86** [comment]
@tylerbutler this piece broke the build-docs pipeline:

image

image

I imagine we need to move the `env` to the task that calls `corepack enable` and that should be enough?

**Pull request merged**